File: java/src/ly/xstream/streaming/Members.java
package ly.xstream.streaming;

import java.util.HashMap;

public class Members {
	HashMap<String,Member> members = new HashMap<String,Member>();
	
	public Member get(String memberId){
		return members.get(memberId);
	}
	
	void add(Member member){
		members.put(member.id, member);
	}
	
	void remove(String memberId){
		members.remove(memberId);
	}
	
	Member socketId(String socketId){
		for(Member m:members.values()){
			if(m.containsSocket(socketId)){
				return m;
			}
		}
		
		return null;
	}
	
	public int getCount(){
		return members.size();
	}
}

File: node.js/lib/members.js
module.exports = Members;

function Members(){
    this.count=0;
    this.members = {};
    this.each = function(func){
        for (key in this.members) {
            func(this.members[key]);
        }
    }
    
    this.get = function(memberId){
        return this.members[memberId];
    }
    
    this.add = function(member){
        this.members[member.id] = member;
        this.count++;
    }
    
    this.remove = function(memberId){
        delete this.members[memberId];
        this.count--;
    }

    this.socketId = function(socketId) {
        for (key in this.members) {
          var member = this.members[key];
            if (member.containsSocket(socketId)) {
                return member;
            }
        }

        return null;
    }
}
